### PR TITLE
S3 proxying & metadata in postgres

### PR DESCRIPTION
**Pull request by meatballhat**
- [x] fill in the rest of this
- [x] no really, this place is a mess

### CODE REVIEW

**Comment by joshk** [comment]
Sooo, I was thinking, we could remove the majority of DB fields and just store the job id, size, and path. At least for now. The rest of the info we can get from the Travis API. Thoughts?

**Comment by joshk** [comment]
\+ content type (thus nice icons)

**Comment by meatballhat** [comment]
That would simplify things greatly, so I'm a fan :smiley_cat:

**Comment by meatballhat** [comment]
oh derp, kept the wrong one.  I know the difference between `job_number` and `job_id`, I swear :wink:

**Comment by joshk** [comment]
Shall we merge in all this awesome work and maybe create a set of issues to discuss the next steps?

File: metadata/metadata.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import (
+	"fmt"
+)
+
+var (
+	errNoMetadata = fmt.Errorf("no metadata available")
+)
+
+// Metadata is the stuff we care about in the metadata database
+type Metadata struct {
```
Comment (joshk) [inline metadata/metadata.go:13]: do we also care about the ID of the record in the DB?

File: server/root_handler.go [hunk 2]
```diff
@@ -5,7 +5,8 @@ import (
 	"net/http"
 )
 
-func (srv *Server) rootHandler(w http.ResponseWriter, r *http.Request, _ map[string]string) {
+func (srv *Server) rootHandler(w http.ResponseWriter, r *http.Request, _ map[string]string) int {
 	w.WriteHeader(http.StatusOK)
 	fmt.Fprintf(w, "sure\n")
+	return http.StatusOK
```
Comment (joshk) [inline server/root_handler.go:11]: maybe we can redirect to the docs?

Comment (meatballhat) [inline server/root_handler.go:11]: :thumbsup:

File: metadata/metadata.go (hunk 1 above)
Comment (meatballhat) [inline metadata/metadata.go:13]: I dunno!  The path is a pretty unique identifier, but there's (currently) no database constraint on it.

**Comment by joshk** [comment]
Yay!

File: metadata/metadata.go (hunk 1 above)
Comment (joshk) [inline metadata/metadata.go:13]: I think we should add the row id as well.

Comment (meatballhat) [inline metadata/metadata.go:13]: :thumbsup:

Comment (meatballhat) [inline metadata/metadata.go:13]: done

File: server/root_handler.go (hunk 2 above)
Comment (meatballhat) [inline server/root_handler.go:11]: I guess I should write some docs :wink:

For now, I added a dumb html thing.

**Pull request merged**